app/models/label/LabelValidationTable.scala

```diff
@@ -158,41 +158,68 @@ object LabelValidationTable {
       case None =>
         // Update val counts in label table if they're not validating their own label and aren't an excluded user.
         if (userThatAppliedLabel != label.userId & !excludedUser)
-          updateValidationCounts(label.labelId, label.validationResult, None)
+          updateValidationCounts(label.labelId, Some(label.validationResult), None)
 
         // Insert a new validation into the label_validation table.
         (validationLabels returning validationLabels.map(_.labelValidationId)) += label
     }
   }
 
+  /**
+   * Deletes a validation in the label_validation table. Also updates validation counts in the label table.
+   */
+  def deleteLabelValidation(label: LabelValidation): Int = db.withTransaction { implicit session =>
+    val oldValidation = validationLabels
+      .filter(x => x.labelId === label.labelId && x.userId === label.userId)
+
+    val excludedUser: Boolean = UserStatTable.userStats.filter(_.userId === label.userId).map(_.excluded).first
+
+    val userThatAppliedLabel: String =
+    labels.filter(_.labelId === label.labelId)
+      .innerJoin(MissionTable.missions).on(_.missionId === _.missionId)
+      .map(_._2.userId)
+      .list.head
+
+    // Delete the old label from the validationLables table.
+    val rowsAffected = oldValidation.delete
+
+    // Update label counts.
+    if (userThatAppliedLabel != label.userId & !excludedUser)
+      updateValidationCounts(label.labelId, None, Some(label.validationResult))
```

Comment:
What you were doing was passing in -1, -2, or -3 as the `newValidationResult` and then subtracting from the validation count using that... But that's already taken care of from the `oldValidationResult` parameter, so you were subtracting by 2 whenever a validation was undone!

I updated it so that instead of passing in a negative number the parameter becomes an `Option[Int]` instead of an `Int`, and we can pass in `None` if the validation is just being deleted since there is no new validation result!

Reply:
Ahh that's my bad! Don't know how I missed that and totally should have tested more after refactoring!